### ISSUE TITLE
Host-side code tests are failing

### ISSUE THREAD

**Issue by eyalroz** [eyalroz/strf#11]
Many (though not most) of the host-side tests are failing:
```
The following tests FAILED:
	  1 - run-test-tr_string-header-only (Child aborted)
	  2 - run-test-tr_string-static-lib (Child aborted)
	  7 - run-test-input_string-header-only (Child aborted)
	  8 - run-test-input_string-static-lib (Child aborted)
	 31 - run-test-width_calculator-header-only (Child aborted)
	 32 - run-test-width_calculator-static-lib (Child aborted)
	 37 - run-test-cstr_writer-header-only (SEGFAULT)
	 38 - run-test-cstr_writer-static-lib (SEGFAULT)
	 39 - run-test-cfile_writer-header-only (SEGFAULT)
	 40 - run-test-cfile_writer-static-lib (SEGFAULT)
	 41 - run-test-streambuf_writer-header-only (SEGFAULT)
	 42 - run-test-streambuf_writer-static-lib (SEGFAULT)
	 43 - run-test-string_writer-header-only (Child aborted)
	 44 - run-test-string_writer-static-lib (Child aborted)
```
You'll notice that failures are in pairs, so it's not a static-lib-vs-headers issue.

Relevant parts of the test log (not pasting the same error twice for each pair):
```
test-tr_string-header-only: malloc.c:2389: sysmalloc: Assertion `(old_top == initial_top (av) && 
old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) &&
((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
...
/home/eyalroz/src/mine/strf/test/input_string.cpp(32): (failure arguments not appeneded) in
function 'int main()'
...
/home/eyalroz/src/mine/strf/test/width_calculator.cpp(61): (failure arguments not appeneded) in
function 'int main()
```
Other failure are silent or with segmentation faults. Need to generate core dumps and get their stacks.

**Comment by robhz786** [comment]
I'm working on this right now. It seems that bug happens when there are too many memory allocations.
I'm removing the uses of std::basic_string in test_utils.hpp, and it seems to solve.
I expect to create a PR soon.

**Comment by robhz786** [comment]
Created PR: https://github.com/eyalroz/strf/pull/15

**Comment by eyalroz** [comment]
@robhz786 : Wow, that's a huge commit!

Can you write a few more lines about the changes you made? Also - was there something wrong with my changes, or did I just trigger that issue you mentioned?

**Comment by robhz786** [comment]
I'm not sure what causes that "malloc.c:2389: sysmalloc:  ..." problem, but it gone away once I removed the use of std::basic_string from the tests.  Later I found an [mistake in oubuf.hpp](https://github.com/eyalroz/strf/pull/15/commits/94f48e96ac9a1f9b518fed545ecc81339a8dfc13), which may be the cause of the bug.

The first commit aimed to adapt the code to CUDA ( like printing the correct messages instead of "failure arguments not appeneded" ). But I also wanted to remove dependency on `boost/lightweight_test.hpp` and reduce the dependency on the OS. 

Another commit removed the use of `std::basic_string` and `std::vector` and things that allocate memory.
 
The remaining commits are small fixes and adjustments.
